### PR TITLE
Feature/hotel class

### PR DESCRIPTION
**Pull request by aliroemhildt**
# What does this PR do?
- write tests for Hotel class
- create Hotel class
- pass all Hotel tests
- refactor a few small things in the other tests/classes/sampleData

# Is this a feature / refactor / bug fix?
feature: Hotel class

# How has this been tested?
- npm test, all tests are passing

# Next steps:
- build out customer dashboard

# Notes for reviewer:
Could you check over the following and let me know if I missed anything? 
- Classes, any missing properties or methods?
- Tests, any missing tests or edge cases?

### CODE REVIEW

File: test/hotel-test.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    expect(hotel.bookings).to.be.an('array');
+    expect(hotel.bookings.length).to.be.greaterThan(0);
+    expect(hotel.bookings).to.deep.equal(allBookings);
+  });
+
+  it('should store all bookings as instances of Booking', () => {
+    expect(hotel.bookings[0]).to.be.an.instanceof(Booking);
+  });
+
+  it('should show all rooms available for a given date', () => {
```
Comment (jasmyn2244) [inline test/hotel-test.js:48]: Could this a case where you test that something unique happens if there are NO rooms available? I think I'd probably handle it in my dom manipulation which is hard to test. But I figured I'd share the thought.

File: test/room-test.js [hunk 2]
```diff
@@ -16,13 +16,13 @@ describe('Room', () => {
   it('should have a number', () => {
```
Comment (jasmyn2244) [inline test/room-test.js:16]: Again, I'm not sure if it's really worth trying, but here you could find a way to make sure that if a data set doesn't have a room number (for some weird reason), that it's doesn't break your code.

**Review by jasmyn2244 — APPROVED** [review]
Hey Ali! I'm so impressed by how clean and simple your filter functions are! The way you use map, reduce, and filter has inspired me to look into refactoring mine before I turn it in :) I'm still learning how to test for edge cases as well, but I threw some ideas out there for you.

File: test/hotel-test.js (hunk 1 above)
Comment (aliroemhildt) [reply test/hotel-test.js:48]: That's a great point! I'm going to add in a test for this method with a date that should return no rooms available.

File: test/room-test.js (hunk 2 above)
Comment (aliroemhildt) [reply test/room-test.js:16]: Another great catch! I'm wondering if that is handled when the api checks the data that is being posted, but I will definitely look into how to make sure that that scenario is being handled somewhere.
